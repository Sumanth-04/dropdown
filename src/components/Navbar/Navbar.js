import React, { Component } from 'react';
import Menu, { Item as MenuItem } from 'rc-menu';
import './index.scss';
import PropTypes from 'prop-types';

export class NavItem extends React.Component {
	static propTypes = {
			navbarStyles: PropTypes.object,
		};
	render() {
		return (
			<MenuItem className={`"rc-menu-item menuItem " ${this.props.className}`} key={this.props.key} {...this.props}>{this.props.children}</MenuItem>
		)
	}
}

export class NavHeader extends React.Component {
	static propTypes = {
		navbarStyles: PropTypes.object,
		};
	render() {
		return (
			<div>{this.props.children}</div>
		)
	}
}

const customizeIndicator = <span>Add More Items</span>;

export default class NavBar extends Component {
	state = {
		overflowedIndicator: undefined,
	};

	/*toggleChildren = () => {
		this.setState(({ children }) => ({
			children: children === children1 ? children2 : children1,
		}));
	};
	*/
	toggleOverflowedIndicator = () => {
		this.setState(({ overflowedIndicator }) => ({
			overflowedIndicator:
				overflowedIndicator === undefined ? customizeIndicator : undefined,
		}));
	};

	render() {
		const { triggerSubMenuAction, children ,navbarStyles} = this.props;
		const { overflowedIndicator } = this.state;

		return (
			<div>
				<Menu
					onClick={this.props.onClick}
					triggerSubMenuAction={triggerSubMenuAction}
					onOpenChange={this.props.onOpenChange}
					selectedKeys={['3']}
					mode={this.props.mode}
					openAnimation={this.props.openAnimation}
					defaultOpenKeys={this.props.defaultOpenKeys}
					overflowedIndicator={overflowedIndicator}
					motion={this.props.motion}
					className={`"rd-menu " ${this.props.className}`}
					style={{...navbarStyles}}
				>
					{children}
				</Menu>
			</div>
		);
	}
}