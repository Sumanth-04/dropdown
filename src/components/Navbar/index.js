import NavBar, { NavItem } from './Navbar';

NavBar.NavMenu = NavBar;
NavBar.NavItem = NavItem;

export default NavBar;