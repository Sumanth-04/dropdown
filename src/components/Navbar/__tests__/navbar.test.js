import React from "react";
import renderer from "react-test-renderer";
import NavBar from "../index";
import { NavItem, NavHeader } from "../Navbar";
//import {isTSAnyKeyword} from '@babel/types';

it("Rendering NavBar", () => {
  const tree = renderer
    .create(
      <NavBar>
        <NavItem navbarStyles={{height : '80px'}}> Redapt Components Library</NavItem>
        <NavItem>
          <br />
        </NavItem>
      </NavBar>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("NavBar", () => {
  const tree = renderer
    .create(
      <NavBar>
        <NavHeader navbarStyles={{height : '80px'}}>React</NavHeader>
        <NavItem>
          NavBar Items
        </NavItem>
        <NavItem><br /></NavItem>
      </NavBar>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});