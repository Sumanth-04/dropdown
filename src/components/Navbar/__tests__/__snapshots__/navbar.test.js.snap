// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar 1`] = `
<div>
  <ul
    className="rc-menu \\"rd-menu \\" undefined rc-menu-root rc-menu-vertical"
    onKeyDown={[Function]}
    onMouseEnter={[Function]}
    onTransitionEnd={[Function]}
    role="menu"
    style={Object {}}
    tabIndex={0}
  >
    <div>
      React
    </div>
    <li
      className="rc-menu-item \\"rc-menu-item menuItem \\" undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
      NavBar Items
    </li>
    <li
      className="rc-menu-item \\"rc-menu-item menuItem \\" undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
      <br />
    </li>
  </ul>
</div>
`;

exports[`Rendering NavBar 1`] = `
<div>
  <ul
    className="rc-menu \\"rd-menu \\" undefined rc-menu-root rc-menu-vertical"
    onKeyDown={[Function]}
    onMouseEnter={[Function]}
    onTransitionEnd={[Function]}
    role="menu"
    style={Object {}}
    tabIndex={0}
  >
    <li
      className="rc-menu-item \\"rc-menu-item menuItem \\" undefined"
      navbarStyles={
        Object {
          "height": "80px",
        }
      }
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
       Redapt Components Library
    </li>
    <li
      className="rc-menu-item \\"rc-menu-item menuItem \\" undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
      <br />
    </li>
  </ul>
</div>
`;
